internal/service/searchindex/resource_search_index_test.go

```diff
@@ -11,18 +11,19 @@ import (
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
 )
 
-func TestAccSearchIndex_basic(t *testing.T) {
+func basicTestCase(tb testing.TB) *resource.TestCase {
```

Comment:
it's great for this example. In some cases we might want to pass projectID so we get the execution project in acc tests but global project in mig,  but that's specific to the tests so no changes need in the infra